### PR TITLE
Feature/geolocation

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

## Changed
- converting from internal geolocation logic to libs abstract class logic.
- updating libs.
- `ES_GEOLOCAITON` global constant to `ES_GEOLOCAITON_IP`.

## Removed
- `es_forms_geolocation_user_location` filter.

## Added
- `es_forms_geolocation_db_location` filter to be able to specify the location of the locations database in your project.
- `es_forms_geolocation_phar_location ` filter to be able to specify the location of the locations database in your project.

### CODE REVIEW

File: src/Geolocation/Geolocation.php [hunk 1]
```diff
@@ -133,7 +193,12 @@ public function isUserGeolocated(string $formId, array $defaultLocations, array
 		}
 
 		// Returns user location retrieved from the API or cookie.
-		$userLocation = $this->getLocationDetails();
+		// USed internal variable for caching optimisations.
```
Comment (iobrado) [inline src/Geolocation/Geolocation.php:196]:
```suggestion
		// Used internal variable for caching optimisations.
```

**Review by iobrado — APPROVED** [review]

**Review by dadadavorin — APPROVED** [review]

File: src/Geolocation/Geolocation.md [hunk 2]
```diff
@@ -10,9 +10,23 @@ Every time a library has a new release we should make an update.
 All releases are listed here: https://github.com/maxmind/GeoIP2-php/releases
 
 Files used:
-* geoip.mmdb
 * geoio.phar
 
+Updated:
+* 2022-08-05 - 2.13.0
+
+## Maxmind GeoLite2 Country db
```
Comment (dingo-d) [inline src/Geolocation/Geolocation.md:18]: Might not be a bad idea to have an action that will check the '_up to dateness_' of the geiop.phar file.

Requests library does this to fetch the certificate bundle when the new ones are available: https://github.com/WordPress/Requests/blob/develop/.github/workflows/update-cacert.yml

What this workflow does is: it checks if the new bundle is available and will automatically create a PR for it. 

#AutomateAllTheThings 😄

**Review by dingo-d — APPROVED** [review]
Left one suggestion, otherwise looks ok 👍🏼

File: src/Geolocation/Geolocation.md (hunk 2 above)
Comment (iruzevic) [reply src/Geolocation/Geolocation.md:18]: pr open

**Pull request merged**